contracts/terraswap_pair/src/contract.rs

```diff
@@ -626,8 +627,10 @@ fn compute_swap(
 
     // offer => ask
     // ask_amount = (ask_pool - cp / (offer_pool + offer_amount)) * (1 - commission_rate)
-    let return_amount: Uint256 =
-        (ask_pool * offer_amount) / Decimal256::from_uint256(offer_pool + offer_amount);
+    let cp: Uint256 = offer_pool * ask_pool;
+    let return_amount: Uint256 = (Decimal256::from_ratio(ask_pool, 1u8)
+        - Decimal256::from_ratio(cp, offer_pool + offer_amount))
+        * Uint256::from(1u8);
```

Comment:
Since the above formula is a simplified formula, it must be rolled back.

Reply:
https://github.com/terraswap/classic-terraswap/pull/14/commits/f0fe4252bf5564b959dca5502ac0d18fd4159529

Reply:
https://github.com/terraswap/classic-terraswap/pull/14/commits/355c95881f7f3df7fe8a2915273e5dd8a8cf15a8

Reply:
let return_amount: Uint256 = (ask_pool * offer_amount) / (offer_pool + offer_amount);

Reply:
https://github.com/terraswap/classic-terraswap/pull/14/commits/73c524189ca36edc2ab13dac275e9c62de933ff9